### DISCUSSION TITLE
Best setup for making targets detect functions from an R data package

### DISCUSSION THREAD

**Discussion by ramiromagno** [ropensci/targets#586] [Help]
Hi Will,

Thanks for developing this wonderful package!

I am a beginner with targets, and I was wondering what is the right setup to track changes to functions developed by myself in an R data package.

My idea is to use targets to develop the somewhat involved workflow for generating several exported datasets and files on disk for this R data package of mine: MyRDataPackage. And I would like those functions that generate these targets to `data-raw/` to be exported functions from the package itself, i.e. I would rather not have them sourced (as in `source("R/functions.R")`) in `_targets.R`.

By reading from [Chapter 6.3 Dependencies ](https://books.ropensci.org/targets/practices.html#dependencies), I got the feeling I could take this approach:

```R
# _targets.R
tar_option_set(envir = getNamespace("MyRDataPackage"))
```

but reading a bit further, namely in [Chapter 6.5 Packages-based invalidation](https://books.ropensci.org/targets/practices.html#packages-based-invalidation), it seems I could also pass my MyRDataPackage to the `imports` argument:

```R
# _targets.R
tar_option_set(
  packages = c("package1", "package2", ...), # `...` is for other packages.
  imports = c("MyRDataPackage")
)
```

So my question is: either approach is fine? Or are there reasons to prefer one over the other?

Many thanks,

Ramiro

**Comment by wlandau** [comment]
Sorry about the inconsistency. Section 6.5 is the current recommendation, section 6.3 is a bit old and no longer the best pattern. Updated the chapter in https://github.com/ropensci-books/targets/commit/a9661e642beb174383222af16c1a599ae10a4735.

**Reply by ramiromagno** [reply]
Thanks for the quick reply!
